### PR TITLE
[AR-1614] build widget shell

### PR DESCRIPTION
**Pull request by karthik-durai**
- Created Widget bubble
- Created Widget IFrame
- Added styles based on provided design
- Created and connected handlers to open and close widget iframe

### CODE REVIEW

File: examples/index.js [hunk 1]
```diff
@@ -5,6 +5,16 @@ const wallet = new WalletProvider({
   appId: "20",
 });
 
+const themeConfig = {
+  assets: {
```
Comment (ajithranka) [inline examples/index.js:9]: See this [comment](https://github.com/arcana-network/wallet/pull/3/files#r813647837). We should organise it as a nested object to make it easier to extend with more orientations or themes.

```
assets:
  logo:
    horizontal:
      dark:
      light:
    vertical:
      dark:
      light:
```

File: src/interfaces.ts [hunk 2]
```diff
@@ -3,6 +3,16 @@ export interface iframeWrapperParams {
...
 
+export interface widgetThemeConfig {
+  assets: {
+    logo_vertical_light: string,
+    logo_vertical_dark: string,
+    logo_horizontal_light: string,
+    logo_horizontal_dark: string,
+  },
+  theme: string
+}
```
Comment (ajithranka) [inline src/interfaces.ts]: - Name as `IThemeConfig` to match the naming convention for other interfaces.
- Use the nesting structure from the above comment.
- Create a new `type Theme = "light" | "dark"` and use it as `theme: Theme`.

File: src/utils.ts [hunk 3]
```diff
@@ -41,4 +42,12 @@ const getAppAddress = async (id: string) => {
...
 
-export { getWalletType }
\ No newline at end of file
+const getWidgetButtonImage = (theme: string, assets: object) => {
+    if(theme === "dark") {
+        return assets.logo_vertical_dark
+    } else if(theme = "light") {
+        return assets.logo_vertical_light
+    }
+}
```
Comment (ajithranka) [inline src/utils.ts]: - Pass `themeConfig: IThemeConfig` as a parameter.
- Check spacing between `if` and conditions.
- Check `theme = "light"`.

File: src/utils.ts [hunk 4]
```diff
@@ -41,4 +42,12 @@ const getAppAddress = async (id: string) => {
...
+
+export { getWalletType, getWidgetButtonImage }
```
Comment (ajithranka) [inline src/utils.ts]: Add an empty line at the end of files.

File: src/utils.ts (hunk 3 above)
Comment (ajithranka) [reply src/utils.ts]: Refactor to `getLogo(themeConfig: IThemeConfig, orientation: Orientation)` to make this more general.

File: src/iframeWrapper.ts [hunk 5]
```diff
@@ -141,15 +145,19 @@ export default class IframeWrapper {
...
-        Object.assign(this.button.style, roundButtonStyle)
+        const buttonImage = document.createElement('img')
+        buttonImage.src = getWidgetButtonImage(theme, assets)
+        this.button.appendChild(buttonImage)
+
+        Object.assign(this.button.style, roundButtonStyle_theme[theme])
         this.button.addEventListener("click", (e) => {
           e.preventDefault();
           this.openFrame();
         });
```
Comment (ajithranka) [inline src/iframeWrapper.ts]: Let's bring our `element` helper here to clean up the DOM creation.

**Review by ajithranka — CHANGES_REQUESTED** [review]

File: src/utils.ts [hunk 6]
```diff
@@ -1,44 +1,61 @@
...
+  let address = json?.address;
+  return address;
+};
+
+const getLogo = (themeConfig: IWidgetThemeConfig, orientation: Orientation) => {
+  const { theme, assets } = themeConfig;
+  return assets.logo[theme][orientation];
+};
+
+const createDomElement = (type: string, props: object, ...children: any) => {
```
Comment (ajithranka) [inline src/utils.ts]: I think we can refine the types further here. Do you mind checking if this works?

```suggestion
const createDomElement = (type: string, props: object, ...children: HTMLElement[]): HTMLElement => {
```

**Review by ajithranka — APPROVED** [review]
This looks great, @karthik-durai! 😄

**Comment by ajithranka** [comment]
@makylfang Adding you to the reviewers list as well. Please do let us know if you spot any issues.

**Review by shrinathprabhu — APPROVED** [review]

**Review by makylfang — APPROVED** [review]

File: src/utils.ts (hunk 6 above)
Comment (karthik-durai) [reply src/utils.ts]: @ajithranka made the suggested changes, let me know your feedback. [commit-link](https://github.com/arcana-network/wallet/pull/6/commits/f092eb90c18113394d1e3d58641ee967b4f4685a)

**Pull request merged**